tests/bot/exts/moderation/test_silence.py

```diff
@@ -268,7 +415,66 @@ async def test_sent_correct_message(self):
             with mock.patch.object(self.cog, "_set_silence_overwrites", return_value=was_silenced):
                 with self.subTest(was_silenced=was_silenced, message=message, duration=duration):
                     await self.cog.silence.callback(self.cog, ctx, duration)
-                    ctx.send.assert_called_once_with(message)
+                    ctx.channel.send.assert_called_once_with(message)
+
+    @mock.patch.object(silence.Silence, "_set_silence_overwrites", return_value=True)
+    @mock.patch.object(silence.Silence, "_force_voice_sync")
+    async def test_sent_to_correct_channel(self, voice_sync, _):
+        """Test function sends messages to the correct channels."""
+        text_channel = MockTextChannel()
+        voice_channel = MockVoiceChannel()
+        ctx = MockContext()
+
+        test_cases = (
+            (None, silence.MSG_SILENCE_SUCCESS.format(duration=10)),
+            (text_channel, silence.MSG_SILENCE_SUCCESS.format(duration=10)),
+            (voice_channel, silence.MSG_SILENCE_SUCCESS.format(duration=10)),
+            (ctx.channel, silence.MSG_SILENCE_SUCCESS.format(duration=10)),
+        )
+
+        for target, message in test_cases:
+            with self.subTest(target_channel=target, message=message):
+                with mock.patch.object(self.cog, "bot") as bot_mock:
+                    bot_mock.get_channel.return_value = AsyncMock()
+                    await self.cog.silence.callback(self.cog, ctx, 10, False, channel=target)
+
+                if ctx.channel == target or target is None:
+                    ctx.channel.send.assert_called_once_with(message)
+
+                else:
+                    ctx.channel.send.assert_called_once_with(message.replace("current channel", target.mention))
+                    if isinstance(target, MockTextChannel):
+                        target.send.assert_called_once_with(message)
+                    else:
+                        voice_sync.assert_called_once_with(target)
+
+            ctx.channel.send.reset_mock()
+            if target is not None and isinstance(target, MockTextChannel):
+                target.send.reset_mock()
+
+    @mock.patch.object(silence.Silence, "_set_silence_overwrites", return_value=True)
+    @mock.patch.object(silence.Silence, "_kick_voice_members")
+    @mock.patch.object(silence.Silence, "_force_voice_sync")
+    async def test_sync_or_kick_called(self, sync, kick, _):
```

Comment:
There should also be a test that neither are called when the target is not a voice channel.